core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,12 +54,33 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).then(function() {
-          // TODO(bhenning): Consolidate the backend rights domain objects and
-          // implement a frontend activity rights domain object. The rights
-          // being passed in here should be used to create one of those objects.
-          if (successCallback) {
-            successCallback();
+        $http.put(collectionRightsUrl, putParams).then(function(response) {
+          // Check if the response from the backend does not contradict
+          // putParams.
+          var isPrivateResponse = (response.data.status === 'private');
+          if (isPrivateResponse === isPublic) {
+            if (isPublic) {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'published, but response.data.is_private returned true.');
+            } else {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'unpublished, but response.data.is_private returned false.');
+            }
+            if (errorCallback) {
+              errorCallback(response.data);
+            }
+          } else {
+            // An additional fetch is required because the GET response object
```

Comment:
It is the most up to date; however, the "schema" from the PUT response and the GET are different. The cache and the logic that depends on the cache (within this domain) relies on the "schema" to be the same; thus, an additional request to GET is needed.